### PR TITLE
Added better documentation for available markdown extensions

### PR DESCRIPTION
**Pull request by aeslaughter**
(refs #6699)

### CODE REVIEW

**Comment by aeslaughter** [comment]
http://mooseframework.org/docs/PRs/36861/site/content/doc_generation/MooseFlavoredMarkdown/

File: python/MooseDocs/extensions/MooseMarkdown.py [hunk 1]
```diff
@@ -45,21 +45,3 @@ def extendMarkdown(self, md, md_globals):
 
 def makeExtension(*args, **kwargs):
     return MooseMarkdown(*args, **kwargs)
-
-if __name__ == '__main__':
```
Comment (dschwen) [inline python/MooseDocs/extensions/MooseMarkdown.py]: Why was this removed?

Comment (aeslaughter) [inline python/MooseDocs/extensions/MooseMarkdown.py]: I was using this for testing, but I think I am beyond needing this now.
